db/revision_cache_lru.go

```diff
@@ -215,7 +215,10 @@ func (rc *LRURevisionCache) getFromCacheByRev(ctx context.Context, docID, revID
 		rc.incrRevCacheMemoryUsage(ctx, value.getItemBytes())
 		// check for memory based eviction
 		rc.revCacheMemoryBasedEviction(ctx)
-		rc.addToHLVMapPostLoad(docID, docRev.RevID, docRev.CV, collectionID)
+		err = rc.addToHLVMapPostLoad(docID, docRev.RevID, docRev.CV, collectionID)
```

Comment:
```suggestion
		err := rc.addToHLVMapPostLoad(docID, docRev.RevID, docRev.CV, collectionID)
```

Reply:
I've intentionally not scoped the error here, essentially if we hit this code here the err is nil, and we if we error here and assign that error to err then below on line 224 we will enter the code block to remove the failed item form the cache altogether. 

I will add comment to code explaining this so its clear.